scripts/PyChop/ISISFermi.py

```diff
@@ -0,0 +1,506 @@
+        ay = bb - ((rat*x2/x0)*gg2*np.cos(gam))
+        a_dd = 1.00/velf
+        v_van_m = am**2  * v_mod
+        v_van_ch = ach**2 * v_ch
+        v_van_jit = ach**2 * v_jit
+        v_van_ya = aya**2 * (wa**2/12.00)
+        v_van_x = ax**2  * v_x
+        v_van_y = ay**2  * v_y
+        v_van_xy = ax*ay  * v_xy
+        v_van_dd = a_dd**2* v_dd
+        # Old version:
+        #v_van = (v_van_m + v_van_ch + v_van_jit + v_van_ya)
+        # New (RAE) version:
+        v_van = (v_van_m + v_van_ch + v_van_jit + v_van_ya + v_van_y + v_van_dd)
+        return v_van
+
+    def getResolution(self, Etrans=None, Ei_in=None, frequency=None):
+        """
+        Calculates resolution (energy) widths
+
+        van = getResolution()
+        van = getResolution(etrans)
+        van = getResolution(etrans, ei, omega)
+
+        Inputs:
+            etrans - list of numpy array of energy transfers to calculate for (meV) [default: linspace(0.05Ei, 0.95Ei, 19)]
+            ei - incident energy in meV [default: preset energy]
+            omega - chopper frequncy in Hz  [default: preset frequency]
+
+        Output:
+            van - the incoherent (Vanadium) energy width at etrans in meV
+        """
+        Ei = self.Ei if Ei_in is None else Ei_in
+        if not Ei:
+            raise ValueError('Incident energy has not been specified')
+        if frequency:
+            oldfreq = self.freq
+            self.setFrequency(frequency)
+        # If not set, sets energy transfers to values to compare exactly to RAE's original implementation.
+        if Etrans is None:
+            Etrans = Ei - np.linspace(0.05*Ei, 0.95*Ei+0.05*0.05*Ei, 19, endpoint=True)
+        convert = 2.3548200
+        x2 = self.__Instruments[self.instname][3]
+        tbin = self.__Instruments[self.instname][9]
+        v_van, tsqmod, tsqchop = self.getVanVar(Ei, frequency, Etrans)
+        if not np.shape(Etrans):
+            Etrans = [Etrans]
+        van = []
+        for ie, en in enumerate(Etrans):
+            van.append(np.real(convert * 8.747832e-4 * np.sqrt((Ei-en)**3) * (np.sqrt(v_van[ie] + (tbin**2/12.00)) * 1.0e6) / x2))
+        if frequency:
+            self.setFrequency(oldfreq)
+        return np.array(van)
+
+    def getFlux(self, Ei_in=None, frequency=None):
+        """
+        Calculates the flux at the sample position in  n / cm**2 . uA.s
+
+        flux = getFlux()
+        flux = getFlux(ei)
+        flux = getFlux(ei, omega)
+
+        Inputs:
+            ei - incident energy in meV [default: preset energy]
+            omega - chopper frequency in Hz  [default: preset frequency]
+
+        Output:
+            flux - the flux in n / cm**2 . uA.s
+        """
+        Ei = self.Ei if Ei_in is None else Ei_in
+        if Ei is None:
+            raise ValueError('Incident energy has not been specified')
+        if frequency:
+            oldfreq = self.freq
+            self.setFrequency(frequency)
+        moderator_flux = self.getMeasuredFlux(Ei)
```

Comment:
> I am not sure the flux calculations are correct.

The flux is taken here from white beam measurements using a calibrated monitor by Rob Bewley. The measurements just keeps going up at high energies / low wavelengths and go out to about 4eV (0.15 Angstrom). I'll ask Rob about the measurements.

If this line is modified to use `getAnalyticFlux` instead, then you see the expected fall off at high energies.